autoscale/src/main/java/com/sequenceiq/periscope/service/ScalingActivityService.java

```diff
@@ -40,6 +51,29 @@ public class ScalingActivityService {
     @Inject
     private RegionAwareCrnGenerator regionAwareCrnGenerator;
 
+    @Inject
+    private AutoscaleRestRequestThreadLocalService restRequestThreadLocalService;
+
+    @Inject
+    private ClusterService clusterService;
+
+    @Override
+    public AuthorizationResourceType getSupportedAuthorizationResourceType() {
+        return AuthorizationResourceType.DATAHUB;
+    }
+
+    @Override
+    @Retryable(value = NotFoundException.class, maxAttempts = 3, backoff = @Backoff(delay = 5000))
+    public String getResourceCrnByResourceName(String clusterName) {
+        return clusterService.findOneByStackNameAndTenant(clusterName, restRequestThreadLocalService.getCloudbreakTenant()).get().getStackCrn();
+    }
+
+    @Override
+    public Optional<String> getEnvironmentCrnByResourceCrn(String resourceCrn) {
+        return Optional.ofNullable(clusterService
+                .findOneByStackCrnAndTenant(resourceCrn, restRequestThreadLocalService.getCloudbreakTenant()).get().getEnvironmentCrn());
```

Comment:
Same here. If the cluster hasn't been synced to periscope yet, or the environmentCrn hasn't been populated into an already created cluster, this will also result in an NPE and cause the subsequent failure of hierarchical authz. Maybe return the envCrn only if populated and return Optional.empty()` otherwise

Reply:
Done